src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageTranslateDialog.vue

```diff
@@ -0,0 +1,295 @@
+		},
+	},
+
+	emits: ['close'],
+
+	data() {
+		return {
+			isMounted: false,
+			selectedFrom: null,
+			selectedTo: null,
+			isLoading: false,
+			translatedMessage: '',
+		}
+	},
+
+	computed: {
+		hasTranslationLanguageDetection() {
+			return this.$store.getters.hasTranslationLanguageDetection()
+		},
+
+		availableLanguages() {
+			return this.$store.getters.getTranslationAvailableLanguages()
+		},
+
+		userLanguage() {
+			return navigator.language.substring(0, 2)
+		},
+
+		sourceTree() {
+			const tree = {}
+			const uniqueSourceLanguages = Array.from(new Set(this.availableLanguages.map(element => element.from)))
+
+			uniqueSourceLanguages.forEach(language => {
+				tree[language] = {
+					id: language,
+					label: this.availableLanguages.find(element => element.from === language)?.fromLabel,
+					translations: this.availableLanguages.filter(element => element.from === language).map(model => ({
+						id: model.to,
+						label: model.toLabel,
+					})),
+				}
+			})
+
+			return tree
+		},
+
+		translationTree() {
+			const tree = {}
+			const uniqueTranslateLanguages = Array.from(new Set(this.availableLanguages.map(element => element.to)))
+
+			uniqueTranslateLanguages.forEach(language => {
+				tree[language] = {
+					id: language,
+					label: this.availableLanguages.find(element => element.to === language)?.toLabel,
+					sources: this.availableLanguages.filter(element => element.to === language).map(model => ({
+						id: model.from,
+						label: model.fromLabel,
+					})),
+				}
+			})
+
+			return tree
+		},
+
+		optionsFrom() {
+			return this.selectedTo?.id
+				? this.translationTree[this.selectedTo?.id]?.sources
+				: Object.values(this.sourceTree).map(model => ({
+					id: model.id,
+					label: model.label,
+				}))
+		},
+
+		optionsTo() {
+			return this.selectedFrom?.id
+				? this.sourceTree[this.selectedFrom?.id]?.translations
+				: Object.values(this.translationTree).map(model => ({
+					id: model.id,
+					label: model.label,
+				}))
+		},
+	},
+
+	watch: {
+		selectedTo() {
+			this.translatedMessage = ''
+		},
+		selectedFrom() {
+			this.translatedMessage = ''
+		},
+	},
+
+	mounted() {
+		this.selectedTo = this.optionsTo.find(language => language.id === this.userLanguage) || null
+
+		if (this.hasTranslationLanguageDetection && this.selectedTo) {
+			this.translateMessage()
+		}
+
+		this.$nextTick(() => {
+			// FIXME trick to avoid focusTrap() from activating on NcSelect
+			this.isMounted = !!this.$refs['translate-modal'].randId
+		})
+	},
+
+	methods: {
+		handleTranslate() {
+			this.translateMessage(this.selectedFrom?.id)
+		},
+
+		async translateMessage(sourceLanguage = null) {
+			try {
+				this.isLoading = true
+				const response = await translateText(this.text, sourceLanguage, this.selectedTo?.id)
+				this.translatedMessage = response.data.ocs.data.text
+			} catch (error) {
+				console.error(error)
+				showError(t('spreed', 'The message could not be translated'))
```

Comment:
The returned error object actually has an error message in the user's language that would be cool to show.

```suggestion
				showError(error.response?.data?.ocs?.data?.message ?: t('spreed', 'The message could not be translated'))
```